### DISCUSSION TITLE
Allow TagsInput suggestions to be refreshed based on input

### DISCUSSION THREAD

**Discussion by mogharsallah** [filamentphp/filament#13570] [Ideas]
Would it be possible to refresh the suggestion list based on the InputTag field's input? This will be extremely helpful when the suggestion list is too large to send at once.
I tried to extend the TagsInput field in such way: 
```php
class SearchableTagsInput extends TagsInput
{
  protected function setUp(): void
  {
    parent::setUp();

    $this->registerListeners([
      'searchable-tags-input::input-change' => [
        function (TagsInput $component, string $statePath, string $value): void {
          $this->suggestions($value ? [ucwords($value)] : []);
        },
      ],
    ]);

    $this->extraInputAttributes(['x-on:keypress.debounce.300ms' => "\$wire.dispatchFormEvent('searchable-tags-input::input-change', '$this->statePath', \$event.target.value)"],
      merge: true);
  }
}

```

This works for the first query only, then something breaks and the page slows down until it crashes.

**Comment by mogharsallah** [comment]
Update:

I was able to get it to work by altering the view dataset element by rendering the available suggestion using livewire and removing the alpinejs conditional rendering
```php
<datalist id="{{ $id }}-suggestions">
  @foreach ($getSuggestions() as $suggestion)
    <option value="{{$suggestion}}"/>
  @endforeach
</datalist>
```

**Comment by drifteaur** [comment]
@danharrin I'm trying to work on this - my site has tens of thousands of tags, loading them all into a suggestions array won't work. I need to search for them as I type.

For the SpatieTagsInput that I'm using, I generated a version that works well for my usecase - admittedly using Claude's help. I'd be happy to clean it up manually, or try using a different approach; could you take a peek at this potential pull request and give me some direction please?

https://github.com/filamentphp/spatie-laravel-tags-plugin/compare/4.x...drifteaur:spatie-laravel-tags-plugin:feature/searchable-tags?expand=1
